### PR TITLE
simplyfied resource wrapping

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: framework/resource/metricsresource/resource.go [hunk 1]
```diff
@@ -181,13 +180,5 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateState inter
...
-	underlying := r.resource
-	for {
-		wrapper, ok := underlying.(internal.Wrapper)
-		if ok {
-			underlying = wrapper.Underlying()
-		} else {
-			return underlying
-		}
-	}
+	return r.resource
```
Comment (xh3b4sd) [inline framework/resource/metricsresource/resource.go]: I do not understand the wrapping magic here. IMO anything implementing `Underlying() framework.Resource` should either return itself or the direct underlying resource. Not more. The next resource will do the same and at some point the last underlying will return itself and we are good. This code here seemed to never return. It was just looping forever which made operators not reconcile anymore.

**Comment by xh3b4sd** [comment]
FYI @rossf7 @fgimenez @kopiczko

**Comment by xh3b4sd** [comment]
See also 

- https://github.com/giantswarm/kvm-operator/pull/406
- https://github.com/giantswarm/aws-operator/pull/776

File: framework/resource/metricsresource/resource.go (hunk 1 above)
Comment (kopiczko) [reply framework/resource/metricsresource/resource.go]: Oh, yeah, there is infinite loop when the underlying resource implements Underlying itself. I didn't anticipate that case.

At the state before  you merged this PR our Resources shouldn't have `Underlying() Resource` method. If you would just remove `Underlying` methods from all resources the thing should work.

I am sorry you had to deal with that. I should have update one of the operators and check that.

**Pull request merged**